kie-server-parent/kie-server-services/kie-server-services-swagger/src/main/java/org/kie/server/services/swagger/SwaggerKieServerExtension.java

```diff
@@ -69,27 +70,34 @@ public void init(KieServerImpl kieServer, KieServerRegistry registry) {
     	this.context = registry;
     	
     	JaxrsScanner jaxrsScanner = new DefaultJaxrsScanner();
-		jaxrsScanner.setPrettyPrint(true);
-		/*
-		 * Set our JAX-RS Scanner with SCANNER_ID_DEFAULT.
-		 * We need to do this before creating the BeanConfig, as this prevents the BeanConfig to register itself as the default scanner.
-		 * The first one wins.
-		 */
-		SwaggerScannerLocator.getInstance()	.putScanner((SwaggerContextService.SCANNER_ID_DEFAULT), jaxrsScanner);
-
-		BeanConfig beanConfig = new BeanConfig();
-		// TODO: Set the API version (or retrieve it from somewhere ...)
-		// beanConfig.setVersion("1.0.2");
-
-		String contextRoot = KieServerEnvironment.getContextRoot();
-		if (contextRoot != null) {
-			beanConfig.setBasePath(contextRoot + "/services/rest");
-		}
-
-		beanConfig.setScan(true);
-    	
+    	jaxrsScanner.setPrettyPrint(true);
+    	/*
+    	 * Set our JAX-RS Scanner with SCANNER_ID_DEFAULT.
+    	 * We need to do this before creating the BeanConfig, as this prevents the BeanConfig to register itself as the default scanner.
+    	 * The first one wins.
+    	 */
+    	SwaggerScannerLocator.getInstance().putScanner((SwaggerContextService.SCANNER_ID_DEFAULT), jaxrsScanner);
+
+    	BeanConfig beanConfig = new BeanConfig();
+		
+    	String contextRoot = KieServerEnvironment.getContextRoot();
+    	if (contextRoot != null) {
+    		beanConfig.setBasePath(contextRoot + "/services/rest");
+    	}
+		
+    	//Set the Info on the Swagger object, not on the BeanConfig ... otherwise the Info on Swagger (which will be 'null') will override the Info we set on the BeanConfig.
+    	beanConfig.getSwagger().setInfo(getInfo());
+    	beanConfig.setScan(true);
+		
     	initialized = true;
     }
+    
+    private Info getInfo() {
+    	Info info = new Info();
+    	info.setTitle("KIE-Server API");
+    	info.setVersion("7.8.0");
```

Comment:
can we remove the version completely?

Reply:
Not really .... it's a required field in the specification: https://swagger.io/specification/#infoObject

The whole point of adding the "Info" object was to be spec compliant ... so removing the version field would defy the purpose.

Also, I still think this should define the version of the API, not the version of KIE-Server. It should indicate versioning of the RESTful interface. So, if we patch stuff -> minor version, if we add stuff or make backwards compatible changes -> minor, and backwards incompatible changes -> major.

Would this be something we could infer from the JAX-RS annotations? For example during build time?

Reply:
in that case I'd say we should use major version of the project/product as the version of the API - in this case version would be set to 7. wdyt?

Reply:
@DuncanDoyle We use revapi for notifications of API changes for example so we can infer it from there. But in general API versioning should be the same as project one I think. All minor releases should or even must be backward-compatible, majors don't need to. So if we stick to this, I agree with @mswiderski.

Reply:
changed the version to be 7.0 - in general refers to major version of the project but uses minor to accommodate unexpected situation